daemon/irc_announce.c

```diff
@@ -38,20 +46,45 @@ static bool announce_channel(const tal_t *ctx, struct ircstate *state, struct pe
 		state->dstate->config.fee_per_satoshi,
 		p->remote.locktime.locktime
 		);
+	sign_privmsg(state, msg);
+	irc_send_msg(state, msg);
+	return true;
+}
 
-	privkey_sign(state->dstate, msg->msg, strlen(msg->msg), &sig);
-	siglen = signature_to_der(state->dstate->secpctx, der, &sig);
-	msg->msg = tal_fmt(msg, "%s %s", tal_hexstr(ctx, der, siglen), msg->msg);
+/* Send an announcement for this node to the channel, including its
+ * hostname, port and ID */
+static void announce_node(const tal_t *ctx, struct ircstate *state)
+{
+	char *hostname = state->dstate->external_ip;
+	int port = state->dstate->portnum;
+	struct privmsg *msg = talz(ctx, struct privmsg);
```

Comment:
hostname should be const char *, port should be u16  (or le16?).
I generally try to avoid talz, so valgrind tells me if I didn't initialzie a field.